modules.d/40network/ifup.sh

```diff
@@ -70,16 +70,17 @@ load_ipv6() {
 }
 
 do_ipv6auto() {
+    local ret
     load_ipv6
     echo 0 > /proc/sys/net/ipv6/conf/$netif/forwarding
     echo 1 > /proc/sys/net/ipv6/conf/$netif/accept_ra
     echo 1 > /proc/sys/net/ipv6/conf/$netif/accept_redirects
     linkup $netif
-    wait_for_ipv6_auto $netif
+    ret=$(wait_for_ipv6_auto $netif)
```

Comment:
err... this doesn't work that way. ```ret=$()```would assign the stdout output of ```wait_for_ipv6_auto``` to ret

You probably want:
```
wait_for_ipv6_auto $netif
ret=$?
```